client/src/components/AppointmentDataForm.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+
+import "react-datetime/css/react-datetime.css";
+
+import { gql, useMutation, useQuery } from "@apollo/client";
+
+import AppointmentForm from "./AppointmentForm";
+import FETCH_APPOINTMENTS from "./fetchAppointments";
+import React from "react";
+import { useHistory } from "react-router-dom";
+
+const CREATE_APPOINTMENT = gql`
+  mutation createAppointment(
+    $startTime: Int!
+    $endTime: Int!
+    $address: String!
+    $note: String!
+  ) {
+    createAppointment(
+      startTime: $startTime
+      endTime: $endTime
+      address: $address
+      note: $note
+    ) {
+      id
+    }
+  }
+`;
+
+const FETCH_APPOINTMENT = gql`
+  query Appointment($id: ID!) {
+    appointment(id: $id) {
+      id
+      startTime
+      endTime
+      location {
+        id
+        address
+      }
+    }
+  }
+`;
+
+const AppointmentDataForm: () => React$Node = (props) => {
+  const history = useHistory();
+  const backToLanding = () => {
+    history.push("/");
+  };
+
+  if (props.match === undefined) {
```

Comment:
Instead of creating <AppointmentForm> twice, you should figure out how to pass a different action into <AppointmentForm> depending on whether it's create or update.

Reply:
Once you have that, you should pass a function that does nothing for the update case so that the page doesn't crash when you click on it.

Reply:
As we just talked this afternoon on Messager, we just leave it the same for now.

Reply:
I just realized that now you have two completely separate components for two URLs, this creates too much duplicate code. You should render the same top level component and define the different logic and sub-components such as a different submit button inside that component.

Reply:
As we talked, it is not an issue now. And I put an empty function for "save draft"button for editing function to come.